module/service.sh

```
@@ -20,6 +20,78 @@ fi
+        echo "gms_user_whitelisted=0" >> "$STOCK_TMP"
+    fi
+
+    # Backup initial GMS AppOps state for surgical restoration on uninstall
+    for op in $GMS_APPOPS; do
+        [ -z "$op" ] && continue
+        op_out=$(cmd appops get com.google.android.gms "$op" 2>/dev/null)
+        op_status=$?
+        op_mode="unknown"
+        if [ "$op_status" -eq 0 ] && [ -n "$op_out" ]; then
+            case "$op_out" in
+                *": allow"*) op_mode="allow" ;;
```

Shellcheck (0.11.0)

[warning] This pattern always overrides a later one on line 79.

(SC2221)